pylearn2/sandbox/nlp/models/mlp.py

```diff
@@ -189,3 +198,303 @@ def get_l1_weight_decay(self, coeff):
         assert isinstance(coeff, float) or hasattr(coeff, 'dtype')
         W, = self.transformer.get_params()
         return coeff * abs(W).sum()
+
+
```

Comment:
It's not necessarily about word2vec, more and more papers are using negative sampling as well. For example Neural Tensor Network is trained with a similar scheme. Quick reference:

> In order to create a testing set for classification, we randomly switch entities from correct testing triplets resulting in a total of 2×#Test triplets with equal number of positive and negative examples.

Socher, R., Chen, D., Manning, C. D., & Ng, A. Y. (2013). Reasoning With Neural Tensor Networks for Knowledge Base Completion. Advances in Neural Information Processing Systems, 926–934.